### PR TITLE
add content for bitglass

### PR DESCRIPTION
**Pull request by spsocprime**
## Proposed Changes

Analytic content for Bitglass

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hey @spsocprime, We have resolved the issue for PR Validations failure and request you to please update your PR branch with the latest changes from **Azure:master** so that the PR Validations can be executed.... Thanks!!!

**Comment by NikTripathi** [comment]
@aprakash13 Please review analytic rules.

**Comment by v-sabiraj** [comment]
There should not be 2 templates with the same ID, but the id **8c8602e6-315d-400f-9d1e-23bbdee1dbfe** is duplicated, please update it.

**Comment by spsocprime** [comment]
some unknown validation error
`##[error]Error reported in diagnostic logs. Please examine the log for more details.
    - /home/vsts/agents/2.196.1/_diag/Worker_20211217-133009-utc.log
System.TimeoutException: The HTTP request timed out after 00:01:40.
 ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
   at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean allowHttp2, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.GetHttpConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.AuthenticationHelper.SendWithAuthAsync(HttpRequestMessage request, Uri authUri, ICredentials credentials, Boolean preAuthenticate, Boolean isProxyAuth, Boolean doRequestAuth, HttpConnectionPool pool, CancellationToken cancellationToken)
   at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Common.VssHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Services.Common.VssHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Common.VssHttpRetryMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.ThrottlingReportHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync(HttpRequestMessage message, HttpCompletionOption completionOption, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.VssHttpClientBase.SendAsync[T](HttpRequestMessage message, Object userState, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Location.Client.LocationHttpClient.GetConnectionDataAsync(ConnectOptions connectOptions, Int64 lastChangeId, CancellationToken cancellationToken, Object userState)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.GetConnectionDataAsync(ConnectOptions connectOptions, Int32 lastChangeId, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.WebApi.Location.VssServerDataProvider.ConnectAsync(ConnectOptions connectOptions, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.Agent.JobServer.ConnectAsync(VssConnection jobConnection)
   at Microsoft.VisualStudio.Services.Agent.Worker.JobRunner.RunAsync(AgentJobRequestMessage message, CancellationToken jobRequestCancellationToken)
   at Microsoft.VisualStudio.Services.Agent.Worker.Worker.RunAsync(String pipeIn, String pipeOut)
   at Microsoft.VisualStudio.Services.Agent.Worker.Program.MainAsync(IHostContext context, String[] args)
,##[error]We stopped hearing from agent Azure Pipelines 5. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610
Started: Today at 16:15`

**Comment by v-sabiraj** [comment]
image
Workbook queries are failing, please look into it, thanks.

**Comment by spsocprime** [comment]
> image Workbook queries are failing, please look into it, thanks.

Seems you don't have Bitglass parser/function in your environment
fixed

**Comment by v-sabiraj** [comment]
image
Please fix the inverted comma's for **Error attempting to login to Bitglass** thanks.

**Comment by v-sabiraj** [comment]
image
Please fix the query, thanks.

**Comment by v-sabiraj** [comment]
@aprakash13 Please review analytic rules. thanks.

**Comment by v-sabiraj** [comment]
Workbooks looks good, can merge once @aprakash13 review and approve, thanks.

**Review by v-sabiraj — DISMISSED** [review]

**Comment by v-sabiraj** [comment]
The query still fails please fix it, thanks.
image

**Review by v-sabiraj — APPROVED** [review]

**Comment by v-sabiraj** [comment]
@aprakash13 Please review and approve analytic rules. thanks.

File: Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+query: |
+  let u_list = Bitglass
+  | where TimeGenerated between(ago(30d) .. ago(1d))
+  | where EventType =~ 'acces'
+  | where EventMessage =~ 'Login'
+  | summarize l_usr = makeset(User)
+  | extend k = 1;
+  Bitglass
+  | where TimeGenerated > ago(24h)
+  | where EventType =~ 'acces'
```
Comment (aprakash13) [inline Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml:23]: Do we need an extra s in access?

File: Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    dataTypes:
+      - Bitglass
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let u_list = Bitglass
+  | where TimeGenerated between(ago(30d) .. ago(1d))
+  | where EventType =~ 'acces'
```
Comment (aprakash13) [inline Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml:17]: Wondering if we need an extra s in access?

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Can we please double check the queries that have join in them, to see if they output desired result? From a cursory look it seems we might need anti joins.

File: Solutions/Bitglass/Analytic Rules/BitglassNewDevice.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+queryFrequency: 1h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  Bitglass
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/Bitglass/Analytic Rules/BitglassNewDevice.yaml:19]: For the content we developed earlier there was requests for changes to remove such `lookbacks`.
image

File: Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml (hunk 1 above)
Comment (spsocprime) [reply Solutions/Bitglass/Hunting Queries/BitglassNewUsers.yaml]: fixed

File: Solutions/Bitglass/Analytic Rules/BitglassImpossibleTravelDistance.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078
+query: |
+  let threshold = 10;
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/Bitglass/Analytic Rules/BitglassImpossibleTravelDistance.yaml]: fixed

File: Solutions/Bitglass/Analytic Rules/BitglassNewAdminUser.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+id: 8c8602e6-315d-400f-9d1e-23bbdee1dbfe
+name: Bitglass - New risky user
+description: |
+  'Detects new risky user.'
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/Bitglass/Analytic Rules/BitglassNewAdminUser.yaml]: fixed

File: Solutions/Bitglass/Analytic Rules/BitglassNewAdminUser.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  Bitglass
+  | where EventType =~ 'admin'
+  | where EventResultDetails	has_all	('User', 'Promoted to administrator')
```
[the remark this answers is not in the document]
Comment (spsocprime) [reply Solutions/Bitglass/Analytic Rules/BitglassNewAdminUser.yaml]: fixed

**Comment by v-sabiraj** [comment]
@aprakash13 can you please approve the changes. Thanks.

**Review by aprakash13 — APPROVED** [review]
Approving the changes.

**Review by v-sabiraj — APPROVED** [review]

**Pull request merged**